File: Web2-Microservices/AirlineMicroservice/Models/Airline.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace AirlineMicroservice.Models
{
    public class Airline
    {
        public Airline()
        {
            Destinations = new HashSet<AirlineDestination>();
            Flights = new HashSet<Flight>();
            Rates = new HashSet<AirlineRate>();
            SpecialOffers = new HashSet<SpecialOffer>();
        }
        public int AirlineId { get; set; }
        public string AdminId { get; set; }

        public string Name { get; set; }
        public string PromoDescription { get; set; }
        public byte[] LogoUrl { get; set; }
        public virtual Address Address { get; set; }

        public virtual ICollection<AirlineDestination> Destinations { get; set; }

        public virtual ICollection<Flight> Flights { get; set; }

        public virtual ICollection<AirlineRate> Rates { get; set; }

        public ICollection<SpecialOffer> SpecialOffers { get; set; }
    }
}

File: Web-SPA/src/app/entities/airline.ts
import { Flight } from './flight';
import { Destination } from './destination';
import { Address } from './address';

export class Airline {

    id: number;
    adminid: number;
    name: string;
    address: Address;
    about: string;
    promoDescription: Array<string>;
    flightDestionations: Array<Destination>;
    flights: Array<Flight>;
    averageRating: number;
    rates: Array<number>;
    logoUrl: string;

    constructor(name: string, address: Address) {
        this.name = name;
        this.address = address;
        this.promoDescription = new Array<string>();
        this.flightDestionations = new Array<Destination>();
        this.flights = new Array<Flight>();
        this.averageRating = 0.0;
        this.rates = new Array<number>();
    }

    rateAriline(rate: number) {
        this.rates.push(rate);
        let sum = 0.00;

        this.rates.forEach(element => {
            sum += element;
        });

        this.averageRating = sum / this.rates.length;
    }
}
